index.js

```diff
@@ -59,15 +60,26 @@ function sendMachines(res, token) {
     var promises = ids.map(function(machineId) {
       return redis.hgetall(machineId)
       .then(function(machine) {
-        machines[machineId] = machine;
+        return redis.hgetall(token + ':' + machineId)
+        .then(function(machineClients) {
+          machine.clients = machineClients;
+          machines[machineId] = machine;
+        });
       });
     });
 
+    promises.push(
+        redis.smembers(token + ':clients')
```

Comment:
We need both. Otherwise there would be a need to constantly check if all machines have all clients. There is no need to use {token}:{machine} key for a client list - it would be enough to do {machineId}_clients. That's a design choice though. By deleting token we're deleting also the client activation choices (or not).